src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -478,7 +669,7 @@ private static unsafe int PickPivotAndPartition(Span<T> keys)
             {
                 if (pivot == null)
                 {
-                    while (Unsafe.IsAddressLessThan(ref leftRef, ref nextToLastRef) && (leftRef = ref Unsafe.Add(ref leftRef, 1)) == null) ;
+                    while (Unsafe.IsAddressLessThan(ref leftRef, ref nextToLastRef) && (leftRef = ref Unsafe.Add(ref leftRef!, 1)) == null) ;
```

Comment:
Why are these changes needed? Looks like a left over from earlier iterations.

Reply:
That looks like a Rider bug that should be fixed in Rider. We should not need to be working around it in the sources.